platform/pwl.c

```diff
@@ -1358,8 +1375,13 @@ pwl_display_destroy (PwlDisplay *self)
     pwl_display_egl_deinit (self);
 
     if (self->display) {
-        display_stop_keyboard_repeat (self);
+        PwlSeat *item, *tmp;
 
+        wl_list_for_each_safe (item, tmp, &self->seats, link) {
+            g_debug ("Seat @ %p: Removed.",
+                     item->seat);
```

Comment:
If you add a `pwl_seat_destroy()` helper function as suggested below, you should move this `g_debug()` call into it, and remove it from here. Also, this is stopping the keyboard repeat timer, not removing the seat, so the debug message is misleading.

Reply:
Fixed.